### PR TITLE
Minor code readability/style refactor

### PR DESCRIPTION
**Pull request by AndyGerlicher**
Some minor changes while reading through code for another PR.

### CODE REVIEW

File: src/Build/BackEnd/BuildManager/BuildParameters.cs [hunk 1]
```diff
@@ -114,10 +105,6 @@ public class BuildParameters : INodePacketTranslatable
         private int _buildId = 0;
 
 #if FEATURE_THREAD_PRIORITY
```
Comment (rainersigwald) [inline src/Build/BackEnd/BuildManager/BuildParameters.cs]: also delete the `#if`?

File: src/Build/Definition/Project.cs [hunk 2]
```diff
@@ -1091,7 +1091,7 @@ public static string GetEvaluatedItemIncludeEscaped(ProjectItemDefinition item)
         ///]
         /// </example>
         /// <remarks>
-        /// Sources of innacuracies: 
+        /// Sources of inaccuracies:
```
Comment (cdmihai) [inline src/Build/Definition/Project.cs]: oops.

File: src/Build/BackEnd/BuildManager/BuildParameters.cs [hunk 3]
```diff
@@ -351,78 +330,69 @@ public ThreadPriority BuildThreadPriority
...
-                {
-                    return new ReadOnlyDictionary<string, string>(new Dictionary<string, string>(0));
-                }
-
-                return new ReadOnlyDictionary<string, string>(_buildProcessEnvironment);
-            }
-        }
+        public IDictionary<string, string> BuildProcessEnvironment => _buildProcessEnvironment == null
+            ? new ReadOnlyDictionary<string, string>(new Dictionary<string, string>(0))
+            : new ReadOnlyDictionary<string, string>(_buildProcessEnvironment);
```
Comment (rainersigwald) [inline src/Build/BackEnd/BuildManager/BuildParameters.cs]:
```c#
new ReadOnlyDictionary<string, string>(_buildProcessEnvironment ?? new Dictionary<string, string>(0));
```
?

Comment (rainersigwald) [inline src/Build/BackEnd/BuildManager/BuildParameters.cs]: Also, this seems confusing, like it should have been a lazy over `_buildProcessEnvironment` but isn't?

Comment (AndyGerlicher) [inline src/Build/BackEnd/BuildManager/BuildParameters.cs]: It seems like it's getting a readonly snapshot. So not Lazy. But still a bit strange to happen on the getter.

Comment (cdmihai) [inline src/Build/BackEnd/BuildManager/BuildParameters.cs]: What about ... an ImmutableDictionary? :) The name of the variable suggests that it does not change much.

**Comment by AndyGerlicher** [comment]
Old compilers :( Will hold off on this until we upgrade them someday.

**Comment by AndyGerlicher** [comment]
I was going to just close this, but we have new compilers! I figured I'd rebase and see if the conflicts weren't too bad. @rainersigwald @cdmihai if you want to take another look.

**Comment by AndyGerlicher** [comment]
@dotnet-bot test Windows_NT Build for Full please

File: src/Build/Definition/ProjectCollection.cs [hunk 4]
```diff
@@ -1607,11 +1600,7 @@ private void ProjectRootElementCache_ProjectDirtiedHandler(object sender, Projec
         /// <param name="e">The event arguments that indicate ProjectRootElement-specific details.</param>
         private void OnProjectXmlChanged(ProjectXmlChangedEventArgs e)
         {
-            var projectXmlChanged = ProjectXmlChanged;
-            if (projectXmlChanged != null)
-            {
-                projectXmlChanged(this, e);
-            }
+            ProjectXmlChanged?.Invoke(this, e);
```
Comment (rainersigwald) [inline src/Build/Definition/ProjectCollection.cs]: Oh good, I've been wanting to do these.

File: src/MSBuild/OutOfProcTaskHostNode.cs [hunk 5]
```diff
@@ -821,14 +821,8 @@ private void RunTask(object state)
...
 
                 // Set culture
-#if FEATURE_CULTUREINFO_SETTERS
-                CultureInfo.CurrentCulture = taskConfiguration.Culture;
-                CultureInfo.CurrentUICulture = taskConfiguration.UICulture;
-
-#else
                 Thread.CurrentThread.CurrentCulture = taskConfiguration.Culture;
                 Thread.CurrentThread.CurrentUICulture = taskConfiguration.UICulture;
-#endif
```
Comment (AndyGerlicher) [inline src/MSBuild/OutOfProcTaskHostNode.cs]: This is interesting. I originally flipped it like the others, but it failed compilation in the 3.5 task host. Looking back through time at the original commit:
https://github.com/Microsoft/msbuild/blob/82177a50da735cc0443ac10fa490d69368403d71/src/XMakeCommandLine/OutOfProcTaskHostNode.cs#L768-L769

This is actually what we had for Dev14. I could do `#IF CLR2`, but it seems like this is as good?

https://referencesource.microsoft.com/#mscorlib/system/globalization/cultureinfo.cs,1f5907384ac6bb55

Thoughts?

**Pull request merged**